Plugin not working on Android 8.0

Hi!

This plugin doesn't work on Android 8.0, logcat outputs this:

```
03-16 18:25:01.899  4884  4884 W zygote  : Attempt to remove non-JNI local reference, dumping thread
03-16 18:25:01.966  4884  4970 E PluginManager: Uncaught exception from plugin
03-16 18:25:01.966  4884  4970 E PluginManager: android.os.FileUriExposedException: file:///storage/emulated/0/Android/data/[MY_APP_ID]/files/instagram8756136426115648363.png exposed beyond app through ClipData.Item.getUri()
03-16 18:25:01.966  4884  4970 E PluginManager:         at android.os.StrictMode.onFileUriExposed(StrictMode.java:1958)
03-16 18:25:01.966  4884  4970 E PluginManager:         at android.net.Uri.checkFileUriExposed(Uri.java:2348)
03-16 18:25:01.966  4884  4970 E PluginManager:         at android.content.ClipData.prepareToLeaveProcess(ClipData.java:941)
03-16 18:25:01.966  4884  4970 E PluginManager:         at android.content.Intent.prepareToLeaveProcess(Intent.java:9735)
03-16 18:25:01.966  4884  4970 E PluginManager:         at android.content.Intent.prepareToLeaveProcess(Intent.java:9741)
03-16 18:25:01.966  4884  4970 E PluginManager:         at android.content.Intent.prepareToLeaveProcess(Intent.java:9720)
03-16 18:25:01.966  4884  4970 E PluginManager:         at android.app.Instrumentation.execStartActivity(Instrumentation.java:1609)
03-16 18:25:01.966  4884  4970 E PluginManager:         at android.app.Activity.startActivityForResult(Activity.java:4472)
03-16 18:25:01.966  4884  4970 E PluginManager:         at org.apache.cordova.CordovaActivity.startActivityForResult(CordovaActivity.java:342)
03-16 18:25:01.966  4884  4970 E PluginManager:         at android.app.Activity.startActivityForResult(Activity.java:4430)
03-16 18:25:01.966  4884  4970 E PluginManager:         at org.apache.cordova.CordovaInterfaceImpl.startActivityForResult(CordovaInterfaceImpl.java:66)
03-16 18:25:01.966  4884  4970 E PluginManager:         at com.vladstirbu.cordova.CDVInstagramPlugin.share(CDVInstagramPlugin.java:127)
03-16 18:25:01.966  4884  4970 E PluginManager:         at com.vladstirbu.cordova.CDVInstagramPlugin.execute(CDVInstagramPlugin.java:73)
03-16 18:25:01.966  4884  4970 E PluginManager:         at org.apache.cordova.CordovaPlugin.execute(CordovaPlugin.java:98)
03-16 18:25:01.966  4884  4970 E PluginManager:         at org.apache.cordova.PluginManager.exec(PluginManager.java:132)
03-16 18:25:01.966  4884  4970 E PluginManager:         at org.apache.cordova.CordovaBridge.jsExec(CordovaBridge.java:57)
03-16 18:25:01.966  4884  4970 E PluginManager:         at org.apache.cordova.engine.SystemExposedJsApi.exec(SystemExposedJsApi.java:41)
03-16 18:25:01.966  4884  4970 E PluginManager:         at org.chromium.base.SystemMessageHandler.nativeDoRunLoopOnce(Native Method)
03-16 18:25:01.966  4884  4970 E PluginManager:         at org.chromium.base.SystemMessageHandler.handleMessage(SystemMessageHandler.java:41)
03-16 18:25:01.966  4884  4970 E PluginManager:         at android.os.Handler.dispatchMessage(Handler.java:105)
03-16 18:25:01.966  4884  4970 E PluginManager:         at android.os.Looper.loop(Looper.java:164)
03-16 18:25:01.966  4884  4970 E PluginManager:         at android.os.HandlerThread.run(HandlerThread.java:65)
03-16 18:25:01.968  4884  4884 W cr_Ime  : updateState: type [0->0], flags [0], show [false], 
```

The problem seems to be easy to solve but Java is not my forte.....
[https://stackoverflow.com/questions/38200282/android-os-fileuriexposedexception-file-storage-emulated-0-test-txt-exposed](https://stackoverflow.com/questions/38200282/android-os-fileuriexposedexception-file-storage-emulated-0-test-txt-exposed)

Thanks

hummmmmmm.
Not sure what I've done but everything was working and I even deployed a new version of my app to the Play Store already.........

Then I basically removed and added cordova back: `ionic cordova platform rm android && ionic cordova platform add android@6.2.3` and did `ionic cordova run android` and the app crashes on launch....

adb logcat:
```
03-27 18:37:31.533  2488  2488 D AndroidRuntime: Shutting down VM
03-27 18:37:31.538  2488  2488 E AndroidRuntime: FATAL EXCEPTION: main
03-27 18:37:31.538  2488  2488 E AndroidRuntime: Process: MY_APP_ID, PID: 2488
03-27 18:37:31.538  2488  2488 E AndroidRuntime: java.lang.RuntimeException: Unable to get provider com.vladstirbu.cordova.plugin.utils.AssetProvider: java.lang.ClassNotFoundException: Didn't find class "com.vladstirbu.cordova.plugin.utils.AssetProvider" on path: DexPathList[[zip file "/data/app/MY_APP_ID-zBBIXHndes8kSG-p2dOqZw==/base.apk"],nativeLibraryDirectories=[/data/app/MY_APP_ID-zBBIXHndes8kSG-p2dOqZw==/lib/arm64, /system/lib64, /system/vendor/lib64]]
03-27 18:37:31.538  2488  2488 E AndroidRuntime:        at android.app.ActivityThread.installProvider(ActivityThread.java:6548)
03-27 18:37:31.538  2488  2488 E AndroidRuntime:        at android.app.ActivityThread.installContentProviders(ActivityThread.java:6094)
03-27 18:37:31.538  2488  2488 E AndroidRuntime:        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6015)
03-27 18:37:31.538  2488  2488 E AndroidRuntime:        at android.app.ActivityThread.-wrap1(Unknown Source:0)
03-27 18:37:31.538  2488  2488 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1756)
03-27 18:37:31.538  2488  2488 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:105)
03-27 18:37:31.538  2488  2488 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:164)
03-27 18:37:31.538  2488  2488 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:6809)
03-27 18:37:31.538  2488  2488 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
03-27 18:37:31.538  2488  2488 E AndroidRuntime:        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
03-27 18:37:31.538  2488  2488 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
03-27 18:37:31.538  2488  2488 E AndroidRuntime: Caused by: java.lang.ClassNotFoundException: Didn't find class "com.vladstirbu.cordova.plugin.utils.AssetProvider" on path: DexPathList[[zip file "/data/app/MY_APP_ID-zBBIXHndes8kSG-p2dOqZw==/base.apk"],nativeLibraryDirectories=[/data/app/MY_APP_ID-zBBIXHndes8kSG-p2dOqZw==/lib/arm64, /system/lib64, /system/vendor/lib64]]
03-27 18:37:31.538  2488  2488 E AndroidRuntime:        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:93)
03-27 18:37:31.538  2488  2488 E AndroidRuntime:        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
03-27 18:37:31.538  2488  2488 E AndroidRuntime:        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
03-27 18:37:31.538  2488  2488 E AndroidRuntime:        at android.app.ActivityThread.installProvider(ActivityThread.java:6533)
03-27 18:37:31.538  2488  2488 E AndroidRuntime:        ... 10 more
```
removing this from the plugin makes it work:

```xml
<config-file target="AndroidManifest.xml" parent="/manifest/application">
        <provider android:authorities="${applicationId}.provider" android:exported="false" android:grantUriPermissions="true" android:name="com.vladstirbu.cordova.plugin.utils.AssetProvider">
          <meta-data android:name="android.support.FILE_PROVIDER_PATHS" android:resource="@xml/instagramplugin_provider_paths" />
        </provider>
    </config-file>
```

but I think that's only because another plugin `cordova-plugin-camera` does basically the same.........
```xml
<config-file target="AndroidManifest.xml" parent="application">
          <provider
              android:name="android.support.v4.content.FileProvider"
              android:authorities="${applicationId}.provider"
              android:exported="false"
              android:grantUriPermissions="true" >
              <meta-data
                  android:name="android.support.FILE_PROVIDER_PATHS"
                  android:resource="@xml/provider_paths"/>
          </provider>
        </config-file>
```

--- a/plugin.xml
+++ b/plugin.xml
@@ -46,6 +46,7 @@
 
 
     <source-file src="src/android/CDVInstagramPlugin.java" target-dir="src/com/vladstirbu/cordova" />
+    <source-file src="src/android/utils/AssetProvider.java" target-dir="src/com/vladstirbu/cordova" />
   </platform>
 
   <!-- ios -->

